module/setter.go

```diff
@@ -128,22 +128,23 @@ if !dstOk && dst.%s != nil {
 	switch {
 	case f.InOneOf():
 		fPath := fmt.Sprintf("%s.(*%s).%s", m.ctx.Name(f.OneOf()), m.ctx.OneofOption(f), fName)
-		buildIndented(buf, tabCount+2, fmt.Sprintf(`if !srcOk && !dstOk {
-	continue
-}
-if srcOk {
+		buildIndented(buf, tabCount+2, fmt.Sprintf(`if srcOk {
 	newSrc = src.%s
 }
 if dstOk {
 	newDst = dst.%s
-} else {
+} else if srcOk {
 	newDst = &%s{}
 	dst.%s = &%s{%s: newDst}
+} else {
```

Comment:
At this point `!srcOk && !dstOk` (which is the original check) - we empty the destination in order to make sure that it matches the source.